### PR TITLE
feat: Cross build for Scala 3

### PR DESCRIPTION
**Pull request by sebastian-alfers**

### CODE REVIEW

File: core/src/main/scala/akka/persistence/r2dbc/query/R2dbcReadJournalProvider.scala [hunk 1]
```diff
@@ -6,12 +6,20 @@ package akka.persistence.r2dbc.query
...
-  override val scaladslReadJournal: scaladsl.R2dbcReadJournal =
+
+  private lazy val scaladslReadJournalInstance =
     new scaladsl.R2dbcReadJournal(system, config, cfgPath)
 
-  override val javadslReadJournal: javadsl.R2dbcReadJournal = new javadsl.R2dbcReadJournal(scaladslReadJournal)
+  override def scaladslReadJournal(): ReadJournal = scaladslReadJournalInstance
+
+  private lazy val javadslReadJournalInstance =
+    new javadsl.R2dbcReadJournal(scaladslReadJournal().asInstanceOf[scaladsl.R2dbcReadJournal])
```
Comment (patriknw) [inline core/src/main/scala/akka/persistence/r2dbc/query/R2dbcReadJournalProvider.scala:21]:
```suggestion
    new javadsl.R2dbcReadJournal(scaladslReadJournalInstance)
```

**Review by patriknw — COMMENTED** [review]
looking good, just one small thing...

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by pvlugter** [comment]
The nightly builds for this are failing: https://github.com/akka/akka-persistence-r2dbc/issues/360

**Pull request merged**